Added initial acceptance test for application submission

stuartmccoll opened the pull request:
This commit contains an initial acceptance test for the successful submission of an application.

I have abstracted methods for each part of the project application form, which will allow us to easily test other paths through this part of the service without re-writing code.

paulmsmith commented:
Running `RSpec` tests reported no failures.
Running `rails test` reported some issues. Hopefully quick to fix.

stuartmccoll commented:
> Running `RSpec` tests reported no failures.
> Running `rails test` reported some issues. Hopefully quick to fix.

Ready for you to take a look at again, @paulmsmith. I've removed all of the Minitest tests and replaced them with RSpec equivalents. No need to run `rails test` anymore, just `rspec`. I've removed the running of `rails test` from the GitHub Action too.

paulmsmith commented:
```
 rspec
...........................................................

Finished in 5.75 seconds (files took 2.42 seconds to load)
59 examples, 0 failures
```
Ran `rspec` locally and no errors.

paulmsmith commented on spec/controllers/released_form_controller_spec.rb:
```diff
@@ -0,0 +1,62 @@
+          id: '2c660111-ab15-4221-98e0-cf0e02748a9b'
+      )
+
+      post :receive, params: {
```
Assuming this is some kind of mock data for the test? :)
